pkg/provisioners/message.go

```diff
@@ -36,12 +39,16 @@ var forwardPrefixes = []string{
 	"x-ot-",
 }
 
+const (
+	// messageHistoryHeader is the header containing all channels in the message history
+	messageHistoryHeader = "knative-message-history"
```

Comment:
Did you want to make this a CloudEvents attribute? If so, there are specific naming conventions (matching the regex `/[0-9a-z]{,20}/`) for CloudEvents attributes:

https://github.com/cloudevents/spec/blob/v0.2/spec.md#attribute-naming-convention

It's probably also worth documenting this field and its format. Right now, this looks like it is a JSON list of Kubernetes ObjectReferences. I'd suggest replacing this with a plain list (either space- or semicolon-separated or JSON) of URLs along the path, which should be somewhat more compact and easier to process without access to the Kubernetes `apiserver`.